lib/livebook_web/plugs/user_plug.ex

```diff
@@ -39,11 +41,27 @@ defmodule LivebookWeb.UserPlug do
     end
   end
 
+  defp ensure_user_identity(conn) do
+    {module, _} = Livebook.Config.identity_provider()
+    identity_data = module.authenticate(LivebookWeb.ZTA, conn)
+
+    if identity_data do
+      identity_data = User.new() |> user_data() |> Map.merge(identity_data)
```

Comment:
I think we want to store just the identity data from `authenticate` and merge with `User.new()` in `ensure_user_data`. Otherwise we will generate a new color every time.

Reply:
Good catch! Thanks!